internal/txlib/add.go

```diff
@@ -47,13 +50,29 @@ func validateFileFilter(input string) error {
 	if res != 1 {
 		return errors.New("you need one <lang> in your File Filter")
 	}
+	if len(filepath.Ext(input)) < 2 {
+		return errors.New("you need to add an extension to your file")
+	}
 	return nil
 }
 
 func validateSourceFile(input string) error {
 	if len(input) < 1 {
 		return errors.New("you need to add a Source File")
 	}
+
+	if len(filepath.Ext(input)) < 2 {
+		return errors.New("you need to add an extension to your Source File")
+	}
+
+	curDir, err := os.Getwd()
+	if err != nil {
+		return errors.New("could not get current path")
+	}
+
+	if _, err := os.Stat(filepath.Join(curDir, input)); os.IsNotExist(err) {
+		return errors.New("you need to add a Source File that exists")
+	}
```

Comment:
This is on me since I wasn't comprehensive before, but maybe:

```suggestion
	if _, err := os.Stat(filepath.Join(curDir, input)) {
		if os.IsNotExist(err) {
			return errors.New("you need to add a Source File that exists")
		} else {
			return errors.New("something went wrong while examining the source file path")
		}
	}
```